Automattic/jetpack
# Fix/connection status inconsistencies

## Proposed changes:

The main purpose of this PR is to unify the connections states and flows in My Jetpack. It should now follow these rules
* If the site is not connected, all "Connection" CTAs should connect the site only, and then redirect users to the pricing page
* If the site is connected, but the user is not, it sends the user to the connection page, then to user connection, and back to My Jetpack, skipping the pricing page

### Other information:


## Jetpack product discussion

GH: 3687b-pb/

## Does this pull request change what data or activity we track or use?

Yes, there is going to be new events
* `jetpack_myjetpack_product_card_fix_site_connection_click`
* `jetpack_myjetpack_product_card_fix_site_connection_success`
For when a user connects their site via a product card

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment

3. Go to `/wp-admin/admin.php?page=my-jetpack` and scroll down to the bottom

4. Select `Connect your site with one click`. Ensure you are scrolled to the top of the page and that after connection, you are directed to the pricing page

https://github.com/user-attachments/assets/412a7cc6-5db8-4bc4-8de2-46546ecb3f8d

5. Disconnect your site and ensure you see a few product cards with the `Connect` CTA
![image](https://github.com/user-attachments/assets/b6adc6e0-5a56-454d-aeba-4a2333c5bd1e)

6. Select Connect and ensure a similar experience to the footer happens where you are connected, brought to the pricing page, and then back to My Jetpack
![image](https://github.com/user-attachments/assets/b8720d3d-cefd-4b82-b34b-06a1ef8398d5)

7. Scroll down to the bottom again and select `Sign In` in the connection footer. On the Connection page, select `Connect your user account`
![image](https://github.com/user-attachments/assets/d1090fc1-e4f3-4d04-92ae-3ee1606a6fe2)

8. Ensure you are led to the user connection flow, and after connecting, you go straight back to the plugin instead of the pricing page

9. Purchase a Jetpack Security plan for your site however you'd like (I used the Store Admin)

10. Go back to My Jetpack and disconnect your site

11. Confirm you see the site connection notice instead of the welcome banner

![image](https://github.com/user-attachments/assets/1f472a9a-8c5d-44d5-83d4-e26c7f64333a)

12. Scroll down to the connection footer and select `Connect`

13. Ensure you have a similar experience where the screen scrolls to the top and connects the site, then directs to the pricing page

14. Go back to My Jetpack and confirm you see the User connection banner now

![image](https://github.com/user-attachments/assets/592bfe0a-15a1-476e-8e2d-e46cdb1b2ff5)

15. Connect your user account any way you'd like (Connect footer, Connection banner, or product card) and ensure the connect flow skips the pricing page

Note: There are other ways to connect I didn't explicitly cover (Such as user connection from the `Activate a license` link), feel free to try out any connection flow you can think of to ensure all site connections and user connections act exactly the same.
